infrastructure/lib/atat-auth-stack.ts

```diff
@@ -36,5 +35,10 @@ export class AtatAuthStack extends cdk.Stack {
         },
       ],
     });
+    const ssmParam = new ssm.StringParameter(this, "UserPoolIdParameter", {
+      description: "Cognito User Pool ID",
+      stringValue: cognitoAuthentication.userPool.userPoolId,
+      parameterName: `/atat/${props.environmentId}/cognito/userpool/id`,
```

Comment:
Is this future proofing for a time when environments are sharing a single SSM store?

Reply:
Yeah so this is actually a little unfortunate. It's actually now-proofing rather than future-proofing. We currently have multiple sandbox environments in an account. Eventually we'll only have one. But we already have other Cognito-related resources that _have_ to be unique globally (the domain specifically) so this is an extension of that. I'd _like_ to find a creative way to make it optional but it's already required and I really want to avoid something like:

```typescript
if (['prod', 'staging', 'dev'].contains(props.environmentId)) {
  // drop the ID from this one use case;
} else {
  // keep it
}
```

Reply:
Got it. Definitely agree with applying KISS here so as to avoid dangerous bugs for special cases around environment names.